src/go/rpk/pkg/api/admin/api_topic_recovery.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package admin
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+// RecoveryRequestParams represents the request body schema for the automated recovery API endpoint.
+type RecoveryRequestParams struct {
+	TopicNamesPattern string `json:"topic_names_pattern"`
+	RetentionBytes    int    `json:"retention_bytes"`
+	RetentionMs       int    `json:"retention_ms"`
+}
+
+type RecoveryStartResponse struct {
+	Code    int    `json:"code"`
+	Message string `json:"message"`
+}
+
+// TopicDownloadCounts represents the count of downloads for a topic.
+type TopicDownloadCounts struct {
+	TopicNamespace      string `json:"topic_namespace"`
+	PendingDownloads    int    `json:"pending_downloads"`
+	SuccessfulDownloads int    `json:"successful_downloads"`
+	FailedDownloads     int    `json:"failed_downloads"`
+}
+
+// TopicRecoveryStatus represents the status of the automated recovery for a topic.
+type TopicRecoveryStatus struct {
+	State           string                `json:"state"`
+	TopicDownloads  []TopicDownloadCounts `json:"topic_download_counts"`
+	RecoveryRequest RecoveryRequestParams `json:"request"`
+}
+
+// StartAutomatedRecovery starts the automated recovery process by sending a request to the automated recovery API endpoint.
+func (a *AdminAPI) StartAutomatedRecovery(ctx context.Context) error {
+	requestParams := &RecoveryRequestParams{
+		TopicNamesPattern: ".*",
```

Comment:
Should this be an input param to `startAutomatedRecovery`?

Reply:
I'm not entirely sure about that at this point. The goal I'm working towards is that we're able to recover a cluster completely as per https://github.com/redpanda-data/cloudv2/issues/4034

My preference would be for the command to recover everything for now and then modify it later when we know more about what the use-cases from a user perspective are

Does that work for you?

Reply:
Changing the API later could hurt backward compatibility, since this is a public repo available for anyone, we might have unknown users.

I'll prefer dealing with this now, what about accepting a string input, and if it's empty default to `".*"`?

Reply:
I was thinking to default to `".*"` when modifying it later. If we're certain that's the way we would like the final interface to look like, I'm okay making that change now :slightly_smiling_face: